openmetadata-service/src/main/java/org/openmetadata/service/events/subscription/SubscriptionPublisher.java

```diff
@@ -0,0 +1,160 @@
+ *   <li>Alerts callbacks that return 3xx are marked as "failed" and no further attempt is made to deliver the events
+ *   <li>Alerts callbacks that return 4xx, 5xx, or timeout are marked as "awaitingRetry" and 5 retry attempts are made
+ *       to deliver the events with the following backoff - 3 seconds, 30 seconds, 5 minutes, 1 hours, and 24 hour. When
+ *       all the 5 delivery attempts fail, the alerts state is marked as "retryLimitReached" and no further attempt is
+ *       made to deliver the events.
+ * </ul>
+ */
+@Slf4j
+public class SubscriptionPublisher extends AbstractEventSubscriptionPublisher {
+  private final CountDownLatch shutdownLatch = new CountDownLatch(1);
+  @Getter private BatchEventProcessor<EventPubSub.ChangeEventHolder> processor;
+  private final EventSubscriptionRepository eventSubscriptionRepository;
+
+  public SubscriptionPublisher(EventSubscription eventSub, CollectionDAO dao) {
+    super(eventSub);
+    this.eventSubscriptionRepository = new EventSubscriptionRepository(dao);
+  }
+
+  @SneakyThrows
+  @Override
+  public void onStart() {
+    setSuccessStatus(System.currentTimeMillis());
+    onStartDelegate();
+    LOG.info("Alert-lifecycle-onStart {}", eventSubscribed.getName());
+  }
+
+  @Override
+  public void onShutdown() {
+    currentBackoffTime = BACKOFF_NORMAL;
+    shutdownLatch.countDown();
+    onShutdownDelegate();
+    LOG.info("Alert-lifecycle-onShutdown {}", eventSubscribed.getName());
+  }
+
+  public synchronized EventSubscription getEventSubscription() {
+    return eventSubscribed;
+  }
+
+  public synchronized void updateEventSubscription(EventSubscription updatedEventSub) {
+    currentBackoffTime = BACKOFF_NORMAL;
+    eventSubscribed.setDescription(updatedEventSub.getDescription());
+    eventSubscribed.setTimeout(updatedEventSub.getTimeout());
+    eventSubscribed.setBatchSize(updatedEventSub.getBatchSize());
+    eventSubscribed.setFilteringRules(updatedEventSub.getFilteringRules());
+    eventSubscribed.setSubscriptionType(updatedEventSub.getSubscriptionType());
+    eventSubscribed.setSubscriptionConfig(updatedEventSub.getSubscriptionConfig());
+  }
+
+  protected synchronized void setErrorStatus(Long attemptTime, Integer statusCode, String reason)
+      throws InterruptedException {
+    SubscriptionStatus status = setStatus(FAILED, attemptTime, statusCode, reason, null);
+    eventSubscriptionRepository.removeProcessorForEventSubscription(eventSubscribed.getId(), status);
+    throw new RuntimeException(reason);
+  }
+
+  protected synchronized void setAwaitingRetry(Long attemptTime, int statusCode, String reason) {
+    setStatus(AWAITING_RETRY, attemptTime, statusCode, reason, attemptTime + currentBackoffTime);
+  }
+
+  protected synchronized SubscriptionStatus setSuccessStatus(Long updateTime) {
+    SubscriptionStatus subStatus =
+        new SubscriptionStatus()
```

Comment:
Move creating SubscriptionStatus to a helper method so that code for creating it is not duplicated.

Reply:
done